ui/src/app/views/workspace/component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import {Component, DebugElement} from '@angular/core';
+import {TestBed, async, fakeAsync, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+import {updateAndTick, simulateInput} from 'testing/test-helpers';
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {CohortsService} from 'generated';
+import {UserService} from 'app/services/user.service';
+import {RepositoryService} from 'app/services/repository.service';
+
+class Context {
```

Comment:
I think I agree with you. I'll be pushing a new version shortly, which initializes the context in the beforeEach. It does require the beforeEach to run with fakeAsync, because it introduces callbacks, but I think that's okay.